lib/AST/GenericSignature.cpp

```diff
@@ -422,7 +422,8 @@ getSubstitutionMap(TypeSubstitutionFn subs,
     Type currentReplacement = depTy.subst(subs, lookupConformance,
                                           SubstFlags::UseErrorType);
     if (auto paramTy = dyn_cast<GenericTypeParamType>(canTy))
-      subMap.addSubstitution(paramTy, currentReplacement);
+      if (!currentReplacement->hasError())
```

Comment:
Do you have a test case that demonstrates where we would hit this that you can add?